### PR TITLE
Json file output

### PR DESCRIPTION
**Pull request by schuelet**
Replaced HDF Handler with JSON Handler because storing in JSON format seems to be simpler.

### CODE REVIEW

File: de.bund.bfr.knime.fsklab.nodes/data/loadJsonIntoPython.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        vars()[targetParam] = pandas.DataFrame.from_records(json_params[sourceParam])
+else:
+    vars()[targetParam] = json_params[sourceParam]
+
+del var_types
+del script_language
+del json_params
+del sourceParam
+del targetParam
+del JSON_FILE_NAME
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/data/loadJsonIntoPython.py:23]: We should have a newline at the end of file.

File: de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java [hunk 2]
```diff
@@ -23,15 +23,16 @@
...
   // the hdf file where all model parameters are stored
-  protected static final String HDF_FILE_NAME = "parameters.h5";  
+  protected static final String JSON_FILE_NAME = "parameters.json";
+  protected static final String JSON_PARAMETERS_NAME = "fsk_parameters";
 
   protected ScriptHandler scriptHandler;
   protected ExecutionContext exec;
   
-  public HDFHandler(ScriptHandler scriptHandler, ExecutionContext exec) {
+  public JsonHandler(ScriptHandler scriptHandler, ExecutionContext exec) {
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java:35]: Constructors of an "abstract" class should not be declared "public" (java:S5993)

Abstract classes should not have public constructors. Constructors of abstract classes can only be called in constructors of their subclasses. So there is no point in making them public. The protected modifier should be enough.

File: de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/PythonJsonHandler.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    compileListOfParameters(fskObj, Parameter.ClassificationEnum.OUTPUT);
+    StringBuilder script = new StringBuilder();
+    script.append("with open('" + JSON_FILE_NAME +"', 'w') as outfile:\n");
+    script.append("\tjson.dump(" + JSON_PARAMETERS_NAME + ", outfile)\n");
+    scriptHandler.runScript(script.toString(), exec, false);
+  }
+
+  @Override
+  public void loadInputParametersFromHDF(FskPortObject fskObj) throws Exception {
+    // TODO THIS IS DUMMY, REMOVE OR RENAME
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/PythonJsonHandler.java:58]: Do we remove this?

File: de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    for (Parameter p : paras) {
+      if (p.getClassification() == classification) {
+        script.append(JSON_PARAMETERS_NAME + "$" + p.getId() + " <- "+ p.getId() + "\n");
+      }
+    }
+    
+    return script.toString();  
+  }
+  
+  private String addTypesToJson(FskPortObject fskObj){
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java:75]: This method should be static.

File: de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    script.append(JSON_PARAMETERS_NAME + "$var_types = var_types\n");
+    return script.toString();
+
+  }
+  @Override
+  public void loadInputParametersFromHDF(FskPortObject fskObj) throws Exception {
+    
+ // TODO THIS IS DUMMY, REMOVE OR RENAME
```
Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java:94]: Do we keep this?

Comment (miguelalba) [inline de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java:94]: Do we keep this?

**Review by miguelalba — CHANGES_REQUESTED** [review]

File: de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java (hunk 5 above)
Comment (schuelet) [reply de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/RJsonHandler.java]: good point. Removed.

**Pull request merged**